src/test-table.md

```diff
@@ -100,6 +104,88 @@ for _, tt := range tests {
 }
 ```
 
+## Avoid Unnecessary Complexity in Table Tests
+
+Table tests can be difficult to read and maintain if the subtests contain conditional 
+assertions or other branching logic. Table tests should **NOT** be used whenever 
+there needs to be complex or conditional logic inside subtests (i.e. logic inside the `for` loop).
+
+Large, complex table tests harm readability and maintainability because test readers may
+have difficulty debugging test failures that occur.
+
+Table tests like this should be split into either multiple test tables or multiple 
+individual `Test...` functions.
+
+Some ideals to aim for are:
+
+* Focus on the narrowest unit of behavior
+* Minimize `if` statements and other control flow (if possible)
```

Comment:
Based on [this comment](https://github.com/uber-go/guide/pull/174#issuecomment-1521829041):

```suggestion
* Minimize "test depth", and avoid conditional assertions (see below)
```

and then expand later, after the list:

> In this context, "test depth" means "within a given test, the number of
> successive assertions that require previous assertions to hold" (similar
> to cyclomatic complexity, except that it also describes prerequisite
> relationships between assertions).
>
> Having "shallower" tests means that there are fewer relationships between
> assertions and, more importantly, that those assertions are less likely
> to be conditional by default.

That might better clarify the actual goal (versus banning conditions in general). I think that this would also allow the guidance to be (1) applied more consistently and (2) more flexible around what a test should look like.

Thoughts?